### PR TITLE
Merge the StorageService2 interface/impl into PayloadStorageService

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Merge StorageService2 into PayloadStorageService and update existing fakes and tests to be compatible with the new fake implementation

### CODE REVIEW

**Comment by bidetofevil** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/1440?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1440** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/1440?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1439** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/1439?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bidetofevil and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-delivery/src/test/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeServiceImplTest.kt [hunk 1]
```diff
@@ -188,14 +190,14 @@ class IntakeServiceImplTest {
             latch.countDown()
         }
         worker.shutdownAndWait(1000)
-        assertEquals(8, payloadStorageService.storedFilenames.size)
+        assertEquals(4, payloadStorageService.storedPayloadCount())
```
Comment (fractalwrench) [inline embrace-android-delivery/src/test/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeServiceImplTest.kt:193]: This indicates a potential bug to me - the intake service has stored 8 payloads with different filenames but the payload storage service only seems to have 4?

Comment (bidetofevil) [reply embrace-android-delivery/src/test/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeServiceImplTest.kt:193]: The file names are based on the metadata, aren't they? So this results in 4 writes and 4 overwrites.

Or is the intention to keep track of the all the versions of the saved payloads?

I think in my version, I was trying to emulate the real file system but in memory. 

I think we can have both if we want. Let me ping you and we can figure out what to do.

**Review by fractalwrench — APPROVED** [review]
LGTM

**Comment by bidetofevil** [comment]
### Merge activity

* **Sep 30, 12:16 PM EDT**: @bidetofevil started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/1440).

**Pull request merged**